test/daemon-test.sh

```diff
@@ -46,25 +60,35 @@ fi
 
 
 # A command (+pattern) that is unlikely to match anything but our own test
-DaemonCmd="$VW -t -i $MODEL --daemon --num_children 1 --quiet --port $PORT"
+DaemonCmd="$VW -t -i $MODEL --daemon $Foreground --num_children 1 --quiet --port $PORT"
 # libtool may wrap vw with '.libs/lt-vw' so we need to be flexible
 # on the exact process pattern we try to kill.
 DaemonPat=`echo $DaemonCmd | sed 's/^[^ ]*vw /.*vw /'`
 
 stop_daemon() {
     # Make sure we are not running. May ignore 'error' that we're not
-    $PKILL -9 -f "$DaemonPat" 2>&1 | grep -q 'no process found'
+    $PKILL -15 -f "$DaemonPat" 2>&1 | grep -q 'no process found'
```

Comment:
Yes, SIGTERM is cleaner/more elegant but unfortunately experience showed that SIGKILL is a more sure thing in some test environments.  It this test there's no concern of leaving anything corrupted so SIGKILL is ok.